Makefile.core.mk

```diff
@@ -97,14 +278,44 @@ prometheus/metrics.proto:
 prometheus/metrics.pb.go: prometheus/metrics.proto
 	@$(PROTOC) $(GOGOFASTER_PLUGIN):. prometheus/metrics.proto
 
-generate: $(googleapis_packages) $(census_packages) prometheus/metrics.pb.go
+k8sapi_prep:
+	@mkdir -p k8sapi/k8s.io/api
+	@cd k8sapi/k8s.io/api && mkdir -p $(k8sapi_packages)
+
+$(k8sapi_protos): %:
+	@curl -sS $(K8SAPI_URL)/$@ -o k8sapi/k8s.io/api/$@.tmp
+	@sed -e '/^option go_package/d' k8sapi/k8s.io/api/$@.tmp > k8sapi/k8s.io/api/$@
+	@rm k8sapi/k8s.io/api/$@.tmp
+
+$(k8sapi_packages): %: k8sapi_prep $(k8sapi_protos)
+	@$(PROTOC) $(GOGOSLICK_PLUGIN):k8sapi -Igoogleprotobuf -Ik8sapimachinery -Ik8sapi k8sapi/k8s.io/api/$@/*.proto
+
+k8sapimachinery_prep:
+	@mkdir -p k8sapimachinery/k8s.io/apimachinery
+	@cd k8sapimachinery/k8s.io/apimachinery && mkdir -p $(k8sapimachinery_packages)
+
+$(k8sapimachinery_protos): %:
+	@curl -sS $(K8SAPIMACHINERY_URL)/$@ -o k8sapimachinery/k8s.io/apimachinery/$@.tmp
+	@sed -e '/^option go_package/d' k8sapimachinery/k8s.io/apimachinery/$@.tmp > k8sapimachinery/k8s.io/apimachinery/$@
+	@rm k8sapimachinery/k8s.io/apimachinery/$@.tmp
+
+$(k8sapimachinery_packages): %: k8sapimachinery_prep $(k8sapimachinery_protos)
+	@$(PROTOC) $(GOGOSLICK_PLUGIN):k8sapimachinery -Igoogleprotobuf -Ik8sapimachinery k8sapimachinery/k8s.io/apimachinery/$@/*.proto
+
+generate: clean $(googleprotobuf_packages) $(googleapis_packages) $(census_packages) prometheus/metrics.pb.go $(k8sapimachinery_packages) $(k8sapi_packages)
+	@mv tmp/opencensus ./opencensus
```

Comment:
I don't understand this move operation.  A generate has a `clean` dependency.  that should clean `/tmp`?

Reply:
It's complicated path problems.

The Go code needs the generated files to be housed at istio.io/gogo-genproto/opencensus. If I build there in the first place, then make gets confused due to things overlapping themselves. So I just moved eveything one level below, and move it back into the right place when done. Icky, but the best I could come up with.